lib/src/cmaf/ham/mapper/hls/mapHamToHls.ts

```diff
@@ -48,55 +48,71 @@ function mapHamToHls(presentation: Presentation[]): Manifest {
 function _generateVideoManifestPiece(videoTrack: VideoTrack) {
 	const newline = `\n`;
 	const mediaSequence = 0; //TODO : save mediaSequence in the model.
-	const manifestToConcat = `#EXT-X-STREAM-INF:BANDWIDTH=${videoTrack.bandwidth},CODECS="${videoTrack.codec}",RESOLUTION=${videoTrack.width}x${videoTrack.height}${newline}${videoTrack.name}${newline}`;
+	const trackFileName = videoTrack.fileName
+		? videoTrack.fileName
+		: `${videoTrack.id}.m3u8`;
+	const manifestToConcat = `#EXT-X-STREAM-INF:BANDWIDTH=${videoTrack.bandwidth},CODECS="${videoTrack.codec}",RESOLUTION=${videoTrack.width}x${videoTrack.height}${newline}${trackFileName}${newline}`;
 	let playlist = videoTrack.segments
 		.map((segment) => {
 			const byteRange =
-				segment.byteRange != undefined
+				segment.byteRange != undefined && segment.byteRange != ''
 					? `#EXT-X-BYTERANGE:${segment.byteRange}${newline}`
 					: '';
-			return `#EXTINF:${segment.duration},${newline}${byteRange}${newline}${segment.url}`;
+			const url = segment.url.includes(' ')
+				? segment.url.replaceAll(' ', '%20')
+				: segment.url;
+			return `#EXTINF:${segment.duration},${newline}${byteRange}${newline}${url}`;
 		})
 		.join(newline);
 	const videoByteRange = videoTrack.byteRange
-		? `#EXT-X-BYTERANGE:${videoTrack.byteRange}${newline}`
+		? `#EXT-X-BYTERANGE:${videoTrack.byteRange.replace('-', '@')}${newline}`
 		: '';
-	playlist = `#EXTM3U${newline}#EXT-X-TARGETDURATION:${videoTrack.duration}${newline}#EXT-X-PLAYLIST-TYPE:VOD${newline}#EXT-X-MEDIA-SEQUENCE:${mediaSequence}${newline}#EXT-X-MAP:URI="${videoTrack.urlInitialization}",${videoByteRange}${newline}${playlist}#EXT-X-ENDLIST`;
+
+	playlist = `#EXTM3U${newline}#EXT-X-TARGETDURATION:${videoTrack.duration}${newline}#EXT-X-PLAYLIST-TYPE:VOD${newline}#EXT-X-MEDIA-SEQUENCE:${mediaSequence}${newline}#EXT-X-MAP:URI="${videoTrack.urlInitialization?.replaceAll(' ', '%20')}",${videoByteRange}${newline}${playlist}${newline}#EXT-X-ENDLIST`;
 
 	return { manifestToConcat, playlist };
 }
 
 function _generateAudioManifestPiece(audioTrack: AudioTrack) {
 	const newline = `\n`;
```

Comment:
This could be a constant. Maybe we can put the aux constant in a separated file.